src/dali/plugin/input/csv/nexo.py

```diff
@@ -0,0 +1,150 @@
+# _CRYPTO_TRANSFER: str = "Crypto Transfer"
+# _DATE: str = "Date"
+# _INTEREST_PAYMENT: str = "Interest Payment"
+# _REFERRAL_BONUS: str = "Referral Bonus"
+# _SOLD_CURRENCY: str = "Sold Currency"
+# _SOLD_QUANTITY: str = "Sold Quantity"
+# _TRADE: str = "Trade"
+# _TRADE_ID: str = "Trade ID"
+# _TYPE: str = "Type"
+# _WITHDRAWAL: str = "Withdrawal"
+# _WITHDRAWAL_FEE: str = "Withdrawal Fee"
+
+
+# types of transactions
+_INTEREST = "Interest"
+_LOCKING_TERM_DEPOSIT = "LockingTermDeposit"
+_UNLOCKING_TERM_DEPOSIT = "UnlockingTermDeposit"
+_FIXED_TERM_INTEREST = "FixedTermInterest"
+_DEPOSIT = "Deposit"
+
+class InputPlugin(AbstractInputPlugin):
+
+    __NEXO: str = "Nexo"
+
+    __TRANSACTION_ID_INDEX = 0
+    __TRANSACTION_TYPE_INDEX = 1
+    __CURRENCY_INDEX: int = 2
+    __AMOUNT_INDEX: int = 3
+
+    __TIMESTAMP_INDEX: int = 7
+
+    __DELIMITER = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        transaction_csv_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__transaction_csv_file: str = transaction_csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__NEXO}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        last_withdrawal_fee: Optional[RP2Decimal] = None
+        with open(self.__transaction_csv_file, encoding="utf-8") as transaction_csv_file:
+            # read CSV with header and skip first row
+            lines = reader(transaction_csv_file)
+
+            # Skip header line
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                transaction_id: str = line[self.__TRANSACTION_ID_INDEX].strip()
+                transaction_type: str = line[self.__TRANSACTION_TYPE_INDEX].strip()
+                currency: str = line[self.__CURRENCY_INDEX].strip()
+                amount = line[self.__AMOUNT_INDEX].strip()
+                timestamp_with_timezone = f"{line[self.__TIMESTAMP_INDEX].strip()} -00:00"
+
+                if transaction_type in [_INTEREST, _FIXED_TERM_INTEREST]:
+                    result.append(
+                        InTransaction(
+                            plugin=self.__NEXO,
+                            unique_id=transaction_id,
+                            raw_data=raw_data,
+                            exchange=self.__NEXO,
+                            holder=self.account_holder,
+                            timestamp=timestamp_with_timezone,
+                            asset=currency,
+
+                            transaction_type=Keyword.INTEREST.name,
+                            # nexo does give us the spot price, but it's often 0 if subcent
```

Comment:
If the price is given and it's > 0 we should use it. If the price is bad (e.g. 0 or missing) we can use `Keyword.UNKNOWN`.

Reply:
The price only has two decimal precision. Maybe I should check for 0 and use `UNKNOWN` in that case, otherwise use the provided price even with only 2 precision? Wdyt?

Reply:
That sounds correct. The Coinbase plugin had the same problem (two decimal precision): see https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/rest/coinbase.py#L723